### PR TITLE
Fix task error report

### LINKED ISSUE #430: Check if ota_state file is valid
https://github.com/edgehog-device-manager/edgehog-device-runtime/blob/17ee1c93d3973df739f0d858dd0c566c15b9a046/src/ota/mod.rs#L644-L658

``` sh
Error: handle for ota-publisher task failed

Caused by:
    InternalError: Unable to publish ota_event: request_uuid is empty
```

### PR DESCRIPTION
**Pull request by harlem88**
close #430

### CODE REVIEW

**Review by joshuachp — CHANGES_REQUESTED** [review]

File: src/ota/mod.rs [hunk 1]
```diff
@@ -353,13 +359,16 @@ where
         if self.ota_status != OtaStatus::Idle {
             error!("ota request already in progress");
 
-            return Err(OtaError::UpdateAlreadyInProgress.into());
```
[the remark this answers is not in the document]
Comment (harlem88) [reply src/ota/mod.rs:356]: Yep, i made mistake with the ota_handler and I changed the return with your suggestion of `IncosistentState`.

File: src/ota/mod.rs [hunk 2]
```diff
@@ -121,6 +121,9 @@ pub enum OtaError {
     /// OTA update aborted by Edgehog half way during the procedure
     #[error("Canceled")]
     Canceled,
+    #[error("InconsistentState")]
+    /// Attempted to start OTA operation while ota status is different from Idle*/
```
Comment (joshuachp) [inline src/ota/mod.rs:125]:
```suggestion
    /// Attempted to start OTA operation while ota status is different from Idle
    #[error("inconsistent ota state")
```

File: src/ota/mod.rs [hunk 3]
```diff
@@ -355,13 +363,12 @@ where
         if self.ota_status != OtaStatus::Idle {
             error!("ota request already in progress");
 
-            return Err(OtaError::UpdateAlreadyInProgress.into());
+            return Err(OtaError::InconsistentState.into());
+        } else {
+            self.ota_status = OtaStatus::Init(msg.ota_id);
+            self.handle_ota_update(msg.cancel).await;
         }
```
Comment (joshuachp) [inline src/ota/mod.rs:370]:
```suggestion
        self.ota_status = OtaStatus::Init(msg.ota_id);
        self.handle_ota_update(msg.cancel).await;
```

File: src/ota/ota_handler.rs [hunk 4]
```diff
@@ -278,6 +279,9 @@ impl From<&OtaError> for OtaStatusMessage {
                 ota_status_message.message = message.to_string()
             }
             OtaError::Canceled => ota_status_message.status_code = "Canceled".to_string(),
+            OtaError::InconsistentState => {
+                ota_status_message.status_code = "InconsistentState".to_string()
```
Comment (joshuachp) [inline src/ota/ota_handler.rs:283]: Are these also defined on the backend? Here I would return an internal error if that's the case

Comment (joshuachp) [reply src/ota/ota_handler.rs:283]: https://docs.edgehog.io/0.9/astarte_interfaces.html#event-statuscode

Here the codes are documented.

```suggestion
                ota_status_message.status_code = "Internal".to_string();
                ota_status_message.message = ota_error.to_string(); 
```

**Review by joshuachp — CHANGES_REQUESTED** [review]

**Review by rgallor — APPROVED** [review]

File: src/ota/ota_handler.rs (hunk 4 above)
Comment (harlem88) [reply src/ota/ota_handler.rs:283]: I agree with you, better to use something that it is already defined also on backed side.

**Review by joshuachp — APPROVED** [review]

**Pull request merged**